### PR TITLE
Create metric filter

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: _docs/MAIN.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
 
 Terraform module to create CloudWatch Log Metric Filters
+
+# variable definitions
+
+As can be seen in the inputs, this module takes the variable `metric_filters`, which is a schema list consisting of the following fields:
+* name (string) - the name for the metric filter
+* filter_pattern (string) - CloudWatch Logs filter pattern for extracting metric data out of ingested log events
+* log_group_name (string) - The name of the log group to associate the metric filter with
+* metric_transformation (map) - A block defining collection of information needed to define how metric data gets emitted
```
Comment (lorengordon) [inline _docs/MAIN.md]: A block in HCL is really a list of maps, so let's just call it a map...
```suggestion
* metric_transformation (map) - A map defining collection of information needed to define how metric data gets emitted
```

File: _docs/MAIN.md [hunk 2]
```diff
@@ -1,3 +1,16 @@
...
+  - The metric_transformation map consists of the following fields:
```
Comment (lorengordon) [inline _docs/MAIN.md]: I think the description is cleaner if this sentence is part of the prior bullet. Then the sub-bullets can be moved up a level. Also, most markdown linters frown on mixing the bullet-style... e.g. `*` to `-`. Just use one or the other throughout.

File: _docs/MAIN.md [hunk 3]
```diff
@@ -1,3 +1,16 @@
 # terraform-aws-tardigrade-cloudwatch-log-metric-filter
 
 Terraform module to create CloudWatch Log Metric Filters
+
+# variable definitions
+
+As can be seen in the inputs, this module takes the variable `metric_filters`, which is a schema list consisting of the following fields:
+* name (string) - the name for the metric filter
```
Comment (lorengordon) [inline _docs/MAIN.md]: This renders a little more clearly, I think, if the variable names are wrapped in backticks...
```suggestion
* `name` (string) - the name for the metric filter
```

File: README.md [hunk 4]
```diff
@@ -2,4 +2,22 @@
 
 Terraform module to create CloudWatch Log Metric Filters
 
+# Variable Definitions
+
+As can be seen in the inputs, this module takes the variable `metric_filters`, which is a schema list consisting of the following fields:
+* `name` (string) - the name for the metric filter
+* `filter_pattern` (string) - CloudWatch Logs filter pattern for extracting metric data out of ingested log events
+* `log_group_name` (string) - The name of the log group to associate the metric filter with
+* `metric_transformation` (map) - A map defining collection of information needed to define how metric data gets emitted The metric_transformation map consists of the following fields:
```
Comment (lorengordon) [inline README.md]:
```suggestion
* `metric_transformation` (map) - A map defining collection of information needed to define how metric data gets emitted. The metric_transformation map consists of the following fields:
```

File: main.tf [hunk 5]
```diff
@@ -1 +1,18 @@
...
+
+  name           = each.value["name"]
+  pattern        = each.value["filter_pattern"]
+  log_group_name = each.value["log_group_name"]
+
+  metric_transformation {
+    name          = each.value["metric_transformation"].name
+    namespace     = each.value["metric_transformation"].namespace
+    value         = each.value["metric_transformation"].value
+    default_value = lookup(each.value["metric_transformation"], "default_value", null)
```
Comment (lorengordon) [inline main.tf]: Due to the use of the `object({...})` for the `metric_transformation` variable, this `default_value` must always be present, so no reason to use `lookup(...)` here...
```suggestion
    default_value = each.value["metric_transformation"].default_value
```

File: README.md (hunk 4 above)
Comment (confusdcodr) [reply README.md]: Done

File: main.tf [hunk 6]
```diff
@@ -1 +1,18 @@
...
+}
 
+resource "aws_cloudwatch_log_metric_filter" "this" {
+  for_each = local.metric_filters
+
+  name           = each.value["name"]
+  pattern        = each.value["filter_pattern"]
+  log_group_name = each.value["log_group_name"]
+
+  metric_transformation {
```
Comment (lorengordon) [inline main.tf]: Hmm, so this actually _is_ a `block` declaration, which is internally represented as a list of maps/objects. Does this resource support multiple `metric_transformation` blocks? If so, we can support that by accepting a list of objects for the variable and using a dynamic block. E.g. is something like this valid?

```
resource "aws_cloudwatch_log_metric_filter" "this" {
  name           = ...
  pattern        = ...
  log_group_name = ...

  metric_transformation {
    name          = ...
    namespace     = ...
    value         = ...
    default_value = ...
  }

  metric_transformation {
    name          = ...
    namespace     = ...
    value         = ...
    default_value = ...
  }
}

Comment (confusdcodr) [reply main.tf]: No, it only supports 1 metric_transformation block

**Pull request merged**